pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -76,3 +75,33 @@ func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, inte
 		return fmt.Errorf("%s not realized", intentPath)
 	})
 }
+
+func (service *RealizeStateService) GetPolicyTier1UplinkPortIP(intentPath string) (string, error) {
+	results, err := service.NSXClient.RealizedEntitiesClient.List(intentPath, nil)
+	err = nsxutil.TransNSXApiError(err)
+	if err != nil {
+		return "", err
+	}
+
+	for _, result := range results.Results {
+		extendAttributes := result.ExtendedAttributes
+		if len(extendAttributes) == 0 {
+			continue
+		}
+		for i := range extendAttributes {
+			if extendAttributes[i].Key != nil && *extendAttributes[i].Key == "IpAddresses" {
+				if len(result.IntentPaths) == 1 {
+					for _, ip := range extendAttributes[i].Values {
```

Comment:
could you move 
if len(result.IntentPaths) == 1  before Line 91?
if if len(result.IntentPaths) != 1 {
 continue
}

Reply:
If extendAttributes[i].Values has multiple values, it seems you will use the first legal one. 
is it possible to have multiple legal value?

Reply:
Good advice, move `len(result.IntentPaths) != 1 ` check before looping the results.extendAttributes

Reply:
>If extendAttributes[i].Values has multiple values, it seems you will use the first legal one.
>is it possible to have multiple legal value?

I feel it should not have multiple values. Here, we just return the first valid value.